src/StapeSGTM.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Stape\Sgtm;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+class StapeSGTM
+{
+    private array $config;
+    private Client $client;
+
+	/**
+	 * @throws SGTMException
+	 */
+	public function __construct(
+		string $gtmServerDomain,
+		string $requestPath = '/data',
+		bool $richsstsse = false,
+		int $protocolVersion = 2)
+    {
+        $this->config = array(
+            'gtm_server_domain' => $gtmServerDomain,
+            'request_path' => $requestPath,
+            'richsstsse' => $richsstsse,
+            'protocol_version' => $protocolVersion
+        );
+
+        $this->client = new Client();
+        $this->validateConfig();
+    }
+
+	/**
+	 * @throws SGTMException
+	 */
+	public function sendEventData(string $eventName, array $eventData = []): bool

```

Comment:
is it possible to replace `array $eventData` with something like `Partial<EventData> $eventData`?